pmda/test/test_util.py

```diff
@@ -130,3 +131,55 @@ def test_make_balanced_slices_ValueError(n_frames, n_blocks,
     with pytest.raises(ValueError):
         make_balanced_slices(n_frames, n_blocks,
                              start=start, stop=stop, step=step)
+
+
+@pytest.mark.parametrize('n_frames', [3, 4, 10, 19, 101, 331, 1000])
+def test_second_order_moments(n_frames):
+    # generate array of random positions in range [-100, 100] for 100 time steps
+    pos = 200*(np.random.random(size=(n_frames, 1000, 3)) - 0.5)
+    # random splitting point
+    isplit = (np.random.randint(1, n_frames-1))
+    # split into two partitions; should be fixed to split by n_blocks partitions
+    p1, p2 = pos[:isplit], pos[isplit:]
+    # create [t, mu, M] lists
+    S1 = [len(p1), p1.mean(axis=0), sumofsquares(p1)]
+    S2 = [len(p2), p2.mean(axis=0), sumofsquares(p2)]
+    # run lists through second_order_moments
+    result = second_order_moments(S1, S2)
+    # compare result to calculations over entire pos array
+    assert result[0] == len(pos)
+    assert_almost_equal(result[1], pos.mean(axis=0))
+    assert_almost_equal(result[2], sumofsquares(pos))
+
+
+@pytest.mark.parametrize('n_blocks', [2, 3, 4, 5, 10, 100, 500])
+@pytest.mark.parametrize('n_frames', [1000, 10000, 50000, 100000])
+def test_reduce(n_frames, n_blocks):
+    # generate array of random positions in range [-100, 100] for 100 time steps
+    pos = 200*(np.random.random(size=(n_frames, 1000, 3)) - 0.5).astype(np.float64)
+    # generate array of indices to split pos by
+    split = []
+    for i in range(n_blocks):
+        split.append(np.random.randint(1, n_frames-1))
+    # remove reduntant indices and sort
```

Comment:
A simple way to generate numbers without replacement (although not super efficient in terms of memory) uses [np.random.shuffle()](https://docs.scipy.org/doc/numpy/reference/random/generated/numpy.random.Generator.shuffle.html#numpy.random.Generator.shuffle) (or Python's `random.shuffle()` but we're already using numpy...)
```python
# all possible indices, except first and last one
indices = range(1, n_frames-1)
# shuffle and take the first n_block ones
split_indices = list(np.random.shuffle(indices)[:n_blocks])
# create start and stop indices for slices
start_indices = [0] + split_indices
stop_indices = split_indices + [n_frames]
# slice "trajectory" pos into blocks
blocks = [pos[i:j] for i,j in zip(start_indices, stop_indices)]

S = [(len(block), block.mean(axis=0, dtype-np.float64), sumofsquares(block)) for block in blocks]
```  
Note how to use list comprehensions to get rid of `for i in range(); List.append(something[i])`.